gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java

```diff
@@ -0,0 +1,533 @@
+  /**
+   * Gets a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param sinkName The resource name of the sink to return.
+   * Example: `"projects/my-project-id/sinks/my-sink-id"`.
+   */
+  public LogSink getSink(String sinkName) {
+    GetSinkRequest request =
+        GetSinkRequest.newBuilder()
+        .setSinkName(sinkName)
+        .build();
+
+    return getSink(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Gets a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public LogSink getSink(GetSinkRequest request) {
+    return getSinkCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Gets a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public ApiCallable<GetSinkRequest, LogSink> getSinkCallable() {
+    return ApiUtils.prepareIdempotentCallable(GET_SINK, settings).bind(channel);
+  }
+
+  // ----- createSink -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param projectName The resource name of the project in which to create the sink.
+   * Example: `"projects/my-project-id"`.
+   *
+   * The new sink must be provided in the request.
+   * @param sink The new sink, which must not have an identifier that already
+   * exists.
+   */
+  public LogSink createSink(String projectName, LogSink sink) {
+    CreateSinkRequest request =
+        CreateSinkRequest.newBuilder()
+        .setProjectName(projectName)
+        .setSink(sink)
+        .build();
+
+    return createSink(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public LogSink createSink(CreateSinkRequest request) {
+    return createSinkCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public ApiCallable<CreateSinkRequest, LogSink> createSinkCallable() {
+    return CREATE_SINK.bind(channel);
+  }
+
+  // ----- updateSink -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates or updates a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param sinkName The resource name of the sink to update.
```

Comment:
I think it is consistent though potentially confusing. The sink must be provided in the request. However, if the sink doesn't already exist on Logging, it is created. It is possible to "update" a nonexistent sink. Tracking in #473 .